### ISSUE TITLE
Calculation bug when using proration and discounts

### ISSUE THREAD

**Issue by mindctrl** [restrictcontentpro/restrict-content-pro#668]
I think this only happens when using proration and discounts together. It affects Stripe and PayPal, and likely the rest. Haven't dug into it deep yet.

See the following series of screenshots to see the difference between what shows on the registration form and what gets calculated for the payment gateway:

#### 1)

![image](https://cloud.githubusercontent.com/assets/1254870/18064518/ca17c2ee-6dfd-11e6-9cae-997a68369097.png)

![image](https://cloud.githubusercontent.com/assets/1254870/18064523/cd8d086c-6dfd-11e6-8182-a598b480f28a.png)

---

#### 2)

![image](https://cloud.githubusercontent.com/assets/1254870/18064592/111a96c6-6dfe-11e6-95df-33b6fbd04b42.png)

![image](https://cloud.githubusercontent.com/assets/1254870/18064597/1736ac84-6dfe-11e6-9d03-9265ef53945d.png)

---

#### 3)

![image](https://cloud.githubusercontent.com/assets/1254870/18064604/1d19594e-6dfe-11e6-92f2-d91a629898a4.png)

![image](https://cloud.githubusercontent.com/assets/1254870/18064607/208aad76-6dfe-11e6-9ea2-b8556b464d9f.png)

**Comment by pippinsplugins** [comment]
Weird . . .

**Comment by mindctrl** [comment]
Probably related: when getting a prorated total of $0, with auto-renew set to never, I'm still seeing a payment option with prevents registration due to trying to pay a $0 amount.

![image](https://cloud.githubusercontent.com/assets/1254870/18145577/d0424340-6f99-11e6-9f90-34786f8eab92.png)

**Comment by mindctrl** [comment]
@pippinsplugins pushed a partial commit to `issue/668`. In my tests, the single payments are accurate. I have not yet tested recurring or made any adjustments there. I wanted to get your thoughts on this route before moving any further with it.

**Comment by pippinsplugins** [comment]
Hmm, not sure I like this.

`$this->amount` should really be the final total, not pre discounts or fees.

**Comment by pippinsplugins** [comment]
If we want a pre-fees / pre-discounts option, let's add `$this->subtotal`.

**Comment by mindctrl** [comment]
Agreed. `$this->amount` already wasn't the final total, as it didn't include the signup fee. It's the subscription price, and used in PayPal to set the `a3` recurring subscription price. I'll work up some changes here, do some testing, and push another commit.

**Comment by pippinsplugins** [comment]
@mindctrl able to work it out with a subtotal?
